File: CS50/pset2/vigenere/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>

int main(int argc, string argv[])

{
    if (argc == 2)
    {
        string f = argv[1];
        int al = strlen(f);
        int z;
        for (z = 0; z < al; z++)
        {
            if (f[z] >= 'A' && f[z] <= 'Z')
            {
                f[z] = f[z] - 'A';
            }
            else if (f[z] >= 'a' && f[z] <= 'z')
            {
                f[z] = f[z] - 'a';
            }
            else
            {
                printf("Usage: ./vigenere k");
                return 1;
            }

        }
        string s = get_string("plaintext: ");
        int i = strlen(s);
        int k;
        int n = 0;

        if (al >= i)
        {
            for (k = 0; k < i; k++)
            {
                //Making sure caption is ok
                if (s[k] >= 'A' && s[k] <= 'Z')
                {
                    s[k] = ((s[k] - 'A' + f[n]) % 26) + 'A';
                    n++;
                }
                else if (s[k] >= 'a' && s[k] <= 'z')
                {
                    s[k] = ((s[k] - 'a' + f[n]) % 26) + 'a';
                    n++;
                }
                else
                {
                    s[k];
                }
            }
        }
        else
        {
            for (k = 0; k < i; k++)
            {
                {
                    //Making sure caption is ok
                    if (s[k] >= 'A' && s[k] <= 'Z')
                    {
                        s[k] = ((s[k] - 'A' + f[n % al]) % ('Z' - 'A' + 1)) + 'A';
                        n++;
                    }
                    else if (s[k] >= 'a' && s[k] <= 'z')
                    {
                        s[k] = ((s[k] - 'a' + f[n % al]) % ('z' - 'a' + 1)) + 'a';
                        n++;
                    }
                    else
                    {
                        s[k];
                    }
                }
            }


        }

        printf("ciphertext: %s\n", s);
    }

    else
    {
        printf("Usage: ./caesar k");
        return 1;
    }
}

File: CS50/pset6/vigenere/vigenere.py
from cs50 import get_string
import sys


def main():
    if len(sys.argv) == 2:
        f = sys.argv[1]
        strlen = len(sys.argv[1])
        key = []

        for k in f:
            if k.isupper():
                t = ord(k) - ord('A')
                key.append(t)
            elif k.islower():
                t = ord(k) - ord('a')
                key.append(t)
            else:
                print("Usage: ./vigenere k")
                exit(1)
        s = get_string("plaintext: ")
        n = 0
        vigenere = []

        for w in range(len(s)):
            # Making sure caption is ok
            if s[w].isupper():
                o = chr(((ord(s[w]) - ord('A') + key[n % strlen]) % 26) + ord('A'))
                n += 1
                vigenere.append(o)
            elif s[w].islower():
                o = chr(((ord(s[w]) - ord('a') + key[n % strlen]) % 26) + ord('a'))
                n += 1
                vigenere.append(o)
            else:
                vigenere.append(s[w])
        print("ciphertext: ", end='')
        print("".join(vigenere))
        exit(0)
    else:
        print("Usage: python vigenere.py [key]")
        exit(1)


if __name__ == "__main__":
    main()